source/driver-vs-mongoose.txt

```diff
@@ -0,0 +1,101 @@
+.. _node-driver-vs-mongoose:
+
+Comparing the Node.js Driver and Mongoose
+=========================================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+
+Overview
+--------
+
+On this page, you can learn about the comparison between
+the official MongoDB Node.js driver and the third-party library called
+Mongoose.
+
+The MongoDB Node.js official driver is a library of functions used to
+connect to and communicate with MongoDB. 
+
+Moongoose is a popular community-developed Object-Document Mapper (ODM)
+library that is built on top of the {+driver-short+}. Though you don't
+need to use an ODM tool to work with MongoDB, some developers prefer
+them as it is similar to using an ORM which they've likely used when
+working with a relational database.
+
+Schema Validation
+-----------------
+
+Both the {+driver-long+} and Mongoose let you apply schema validation
+to your collection. However, the biggest difference between the two are
+the way they each set and manage schema validation.
+
+The {+driver-long+} applies the schema validation at the database
+level. In comparison, Mongoose applies schema validation at the application level. In the
+{+driver-short+} schema validation works on individual collections by
+using either `JSON Schema Validation <https://www.mongodb.com/docs/manual/core/schema-validation/specify-json-schema/>`__
+or `Schema Validation with Query Operators <https://www.mongodb.com/docs/manual/core/schema-validation/specify-query-expression-rules/>`__.
+Validation with query operators is useful for dynamic validation rules
+calculated at runtime. By contrast, JSON validation is useful for, among other
+reasons, being an open standard which makes it easy to connect to
+third-party libraries.
+
+Mongoose applies the schema validation in a twofold procedure; firstly by telling your
+application what data types and fields are allowed to be in a particular
+collection, then it confirms the data matches the set criteria. Mongoose's
+schema validation works on the application level, meaning MongoDB
+doesn't know it exists.
```

Comment:
This reads a little weird. Maybe something like:

_"Unlike the {+driver-long+}, Mongoose's schema validation occurs client-side. Schema validation is applied by first telling your application what data types and fields are allowed to be in a particular collection, then by confirming the data matches the set criteria."_